src/plots/plots.js

```diff
@@ -798,7 +798,7 @@ plots.supplyDataDefaults = function(dataIn, dataOut, layout, fullLayout) {
 
     for(i = 0; i < dataIn.length; i++) {
         trace = dataIn[i];
-        fullTrace = plots.supplyTraceDefaults(trace, cnt, fullLayout, i);
+        fullTrace = plots.supplyTraceDefaults(trace, cnt, fullLayout, i, dataIn.length);
```

Comment:
I would prefer, stashing `dataIn.length` on `fullLayout` then passing it around like this.

Something like `fullLayout._dataInLength` that could be set [here](https://github.com/plotly/plotly.js/blob/b2e54861dfe9960a7bb447f0a622d4198a4e22c7/src/plots/plots.js#L453) side-by-side to `fullLayout._dataLength`.